platform/darwin/src/MGLAccountManager.m

```diff
@@ -102,20 +112,10 @@ + (NSURL *)apiBaseURL {
 
 #if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
 
-+ (void)setSkuToken:(NSString *)skuToken {
-    MGLAccountManager.sharedManager.skuTokenExpiration = [NSDate dateWithTimeIntervalSinceNow:MGLAccountManagerSKUTokenLifespan];
-    MGLAccountManager.sharedManager.skuToken = skuToken;
-}
-
 + (NSString *)skuToken {
-    return [MGLAccountManager.sharedManager isSKUTokenExpired] ?
-        [MBXSKUToken tokenForSKUID:MBXAccountsSKUIDMaps type:MBXAccountsSKUTypeUser] :
-        MGLAccountManager.sharedManager.skuToken;
-}
-
-- (BOOL)isSKUTokenExpired {
-    NSTimeInterval secondsUntilExpiration = [MGLAccountManager.sharedManager.skuTokenExpiration timeIntervalSinceNow];
-    return secondsUntilExpiration < 0;
+    return MGLAccountManager.sharedManager.useExternalAccountManager ?
+        [MGLAccountManager.sharedManager.externalAccountsClass performSelector:MGLAccountManager.sharedManager.externalSKUTokenMethod] :
```

Comment:
The warning about the selector being unknown is probably because `externalSKUTokenMethod` is neither static nor constant. Does it persist if we inline `NSSelectorFromString(MGLAccountManagerExternalMethodName)` here? `NSSelectorFromString()` is very fast.

If the issue is also that `externalAccountsClass` is computed dynamically, then you might be able to drop down to Objective-C runtime functions like `objc_msgSend()`.

Reply:
Inlining would simplify the implementation and fix the warning about a potential leak. Let's give that a try.

Reply:
This (extremely unsafe looking) bit of code still prompts the "PerformSelector may cause a leak because its selector is unknown” warning:

```objc
[NSClassFromString(MGLAccountManagerExternalClassName) performSelector:NSSelectorFromString(MGLAccountManagerExternalMethodName)]
```

Will poke at this more.

Reply:
In https://github.com/mapbox/mapbox-gl-native/pull/14802/commits/af64bce82491603efb1da00f34a2f7a09e6e7ef0 I went with an approach [outlined here](https://stackoverflow.com/a/20058585/2094275) — it gives the compiler more context about the selector call. What’cha think, @frederoni @1ec5?

Alternatively, it is possible (though evidently discouraged) to disable the leak warning using `#pragma clang diagnostic ignored “-Warc-performSelector-leaks”`.

Reply:
Tested https://github.com/mapbox/mapbox-gl-native/commit/af64bce82491603efb1da00f34a2f7a09e6e7ef0 by changing the value of `MGLAccountManagerExternalClassName` to the existing SKU token class (`MBXSKUToken`) — it functioned as expected. This seems like it will work, but we will obviously need to test with a differently-named accounts class that is not present in the maps SDK.